src/ReusableRelayBuilder.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace Relay;
+
+use ArrayObject;
+use InvalidArgumentException;
+
+class ReusableRelayBuilder
+{
+    protected $resolver;
+
+    public function __construct($resolver = null)
+    {
+        $this->resolver = $resolver;
+    }
+
+    public function newInstance($queue)
+    {
+        return new ReusableRelay(new RelayFactory(
```

Comment:
I worry about the `new RelayFactory` here because I don't see a way to replace it with a subtype.